dm/worker/subtask.go

```diff
@@ -370,10 +369,19 @@ func (st *SubTask) setResult(result *pb.ProcessResult) {
 }
 
 // Result returns the result of the sub task
+// Note this method will omit the `Error` field in `pb.ProcessError`, so no duplicated
+// error message information will be displayed in `query-status`, as the `Msg` field
+// contains enough error information.
 func (st *SubTask) Result() *pb.ProcessResult {
 	st.RLock()
 	defer st.RUnlock()
-	return proto.Clone(st.result).(*pb.ProcessResult)
+	result := proto.Clone(st.result).(*pb.ProcessResult)
+	if result != nil {
+		for i := range result.Errors {
+			result.Errors[i].Error = nil
+		}
+	}
+	return result
```

Comment:
these code is same with `realRelayHolder.Result()`, how about define a funcion to do this?

Reply:
refined, PTAL